x-pack/plugins/security_solution/public/common/components/event_details/table/use_action_cell_data_provider.ts

```diff
@@ -153,5 +156,27 @@ export const useActionCellDataProvider = ({
     linkValue,
     values,
   ]);
+  if (alertsOnly && cellData) {
+    const ensureOnlyRuleProviders: DataProvider = {
+      enabled: true,
+      id: ALERT_RULE_UUID,
+      name: ALERT_RULE_UUID,
+      excluded: false,
+      kqlQuery: '',
+      queryMatch: {
+        field: ALERT_RULE_UUID,
+        value: '*',
+        operator: EXISTS_OPERATOR,
+      },
+    };
+    cellData.dataProviders = cellData?.dataProviders.map((provider) => {
```

Comment:
nit: the optional `?.` chaining in `cellData?.dataProviders.map((provider)` is not required because `cellData` is within scope of the `if (alertsOnly && cellData)` check.

Reply:
Is it possible to avoid mutating `cellData` here after it was initialized in the

```ts
const cellData = useMemo(() => {
  // ...
}
```

block above it? (There's already a lot of action in the `useMemo` above, which may have influenced the decision to mutate it here, but it might still be preferable to avoid mutating it here.)